elasticsearch/client.go

```diff
@@ -1,26 +1,81 @@
 package elasticsearch
 
-// Client provides methods to wrap around the elastcsearch package in order to facilitate unit testing
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	rchttp "github.com/ONSdigital/dp-rchttp"
+)
+
+// Client represents an instance of the elasticsearch client
 type Client struct {
-	URL string
+	url          string
+	rchttpClient rchttp.Clienter
 }
 
 // New creates a new elasticsearch client
-func New(URL string) *Client {
-
-	Setup(URL)
-
+func New(url string, rchttpClient rchttp.Clienter) *Client {
 	return &Client{
-		URL: URL,
+		url:          strings.TrimRight(url, "/"),
+		rchttpClient: rchttpClient,
 	}
 }
 
 // Search is a method that wraps the Search function of the elasticsearch package
-func (cli *Client) Search(index string, docType string, request []byte) ([]byte, error) {
-	return Search(index, docType, request)
+func (cli *Client) Search(ctx context.Context, index string, docType string, request []byte) ([]byte, error) {
+	return cli.post(ctx, index, docType, "_search", request)
 }
 
 // MultiSearch is a method that wraps the MultiSearch function of the elasticsearch package
-func (cli *Client) MultiSearch(index string, docType string, request []byte) ([]byte, error) {
-	return MultiSearch(index, docType, request)
+func (cli *Client) MultiSearch(ctx context.Context, index string, docType string, request []byte) ([]byte, error) {
+	return cli.post(ctx, index, docType, "_msearch", request)
+}
+
+// GetStatus makes status call for healthcheck purposes
+func (cli *Client) GetStatus() ([]byte, error) {
+	req, err := http.NewRequest("GET", cli.url+"/_cat/health", nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := cli.rchttpClient.Do(context.Background(), req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	response, err := ioutil.ReadAll(resp.Body)
```

Comment:
I would do:
```
response, err := ioutil.ReadAll(resp.Body)
if err != nil {
    return nil, errors.Wrap(err, "elaticsearchClient error reading get status response body")
}
```
As this gives the error a very clear context of the scenario and will be über helpful when debugging.

Reply:
OK, also added it to the similar call in the post() method too.